ws-server/src/server.rs

```diff
@@ -271,17 +274,25 @@ async fn background_task(
 	while !stop_server.shutdown_requested() {
 		data.clear();
 
-		if let Err(e) = method_executors.select_with(receiver.receive_data(&mut data)).await {
-			tracing::error!("Could not receive WS data: {:?}; closing connection", e);
-			tx.close_channel();
-			return Err(e.into());
-		}
-
-		if data.len() > max_request_body_size as usize {
-			tracing::warn!("Request is too big ({} bytes, max is {})", data.len(), max_request_body_size);
-			send_error(Id::Null, &tx, ErrorCode::OversizedRequest.into());
-			continue;
-		}
+		if let Err(err) = method_executors.select_with(receiver.receive_data(&mut data)).await {
+			match err {
+				SokettoError::Closed => {
+					tracing::info!("Remote peer terminated the connection: {}", conn_id);
+					tx.close_channel();
+					return Ok(());
+				}
+				SokettoError::MessageTooLarge { current, maximum } => {
+					tracing::warn!("Request is too big ({} bytes, max is {})", current, maximum);
+					send_error(Id::Null, &tx, ErrorCode::OversizedRequest.into());
+				}
+				// NOTE: io::Error might happen if the remove peer terminated connection.
+				e @ _ => {
+					tracing::error!("WS recv error: {:?} => terminate connection {}", e, conn_id);
```

Comment:
should we remove downgrade this error log?

It might confuse some people

Reply:
My preference would be to match explicitly on all error variants. 

Looking at the various code paths behind `receive_data` it seems like this could be a few different error variants, mostly IO errors (which I think are rare but probably pretty catastrophic), but also utf8 parsing errors, and a few other instances of bad/corrupted control data.

I think we should continue logging this, but perhaps more granularly?

Reply:
we get this `Oct 21 09:58:46.309 ERROR jsonrpsee_ws_server::server: WS recv error: Io(Os { code: 107, kind: NotConnected, message: "Transport endpoint is not connected" }) => terminate connection 0`

sometimes when the remote peer closes the connection....